src/httpserver.cpp

```diff
@@ -457,21 +475,21 @@ void StopHTTPServer()
         delete workQueue;
         workQueue = nullptr;
     }
+    {
+        LogPrint(BCLog::HTTP, "Disabling reading on current connections\n");
```

Comment:
Can you add a comment about what the effect of disabling reading on the current connections is?

If this just disables reading and doesn't close the connections or interrupt pending writes, it seems like shutdown code could potentially get stuck forever (if a client isn't reading and there is a pending write and data is backed up).

Reply:
So the event base quits when there are no active or pending events. Writing the response is an active event which will go away at the end. Reading can be a pending event (for instance, the client sends stop RPC but keeps the connection open — which happens in the functional tests) so that will prevent the base to quit automatically.